File: maxxieminnie.jl
using Combinatorics

function swap(x, i, j)
    y = copy(x)
    y[i], y[j] = y[j], y[i]
    y
end

function maxmin(x)
    nvec = parse.(Int64, split(string(x), ""))
    opts = collect(combinations(1:length(nvec), 2))
    new = [[nvec]; map(x -> swap(nvec, x...), opts)]
    keep = filter(x -> x[1] != 0, new)
    vals = parse.(Int64, join.(keep))
    [maximum(vals) minimum(vals)]
end

maxmin(213)
maxmin(12345)
maxmin(100)
maxmin(11321)

File: maxxieminnie.py
import itertools

def swap(x, t):
    y = x.copy()
    i, j = t
    y[i], y[j] = y[j], y[i]
    return y

def minmax(num): 
    nums = [int(i) for i in str(num)]
    opts = itertools.combinations(range(len(nums)), 2)
    new = map(lambda x: swap(nums, x), list(opts))
    keeps = list(filter(lambda x: x[0] != 0, new))
    keeps.append(nums)
    vals = list(map(lambda x: int(''.join(map(str, x))), keeps))
    return (max(vals), min(vals))

minmax(213)
minmax(12345)
minmax(100)
minmax(11321)
